drake/util/drakeGeometryUtil.h

```diff
@@ -96,48 +153,241 @@ Eigen::Quaternion<typename Derived::Scalar> quat2eigenQuaternion(const Eigen::Ma
  * axis2x
  */
 template <typename Derived>
-DLLEXPORT Eigen::Vector4d axis2quat(const Eigen::MatrixBase<Derived>& a);
+Eigen::Vector4d axis2quat(const Eigen::MatrixBase<Derived>& a) {
+  EIGEN_STATIC_ASSERT_VECTOR_SPECIFIC_SIZE(Eigen::MatrixBase<Derived>, 4);
+  auto axis = a.template head<3>();
+  auto angle = a(3);
+  auto arg = 0.5 * angle;
+  auto c = std::cos(arg);
+  auto s = std::sin(arg);
+  Eigen::Vector4d ret;
+  ret << c, s * axis;
+  return ret;
+}
 
 template <typename Derived>
-DLLEXPORT Eigen::Matrix<typename Derived::Scalar, 3, 3> axis2rotmat(const Eigen::MatrixBase<Derived>& a);
+Eigen::Matrix<typename Derived::Scalar, 3, 3> axis2rotmat(const Eigen::MatrixBase<Derived>& a) {
+  EIGEN_STATIC_ASSERT_VECTOR_SPECIFIC_SIZE(Eigen::MatrixBase<Derived>, 4);
+  const auto& axis = (a.template head<3>())/(a.template head<3>()).norm();
+  const auto& theta = a(3);
+  auto x = axis(0);
+  auto y = axis(1);
+  auto z = axis(2);
+  auto ctheta = std::cos(theta);
+  auto stheta = std::sin(theta);
+  auto c = 1 - ctheta;
+  Eigen::Matrix<typename Derived::Scalar, 3, 3> R;
+  R <<
+  ctheta + x * x * c , x * y * c - z * stheta, x * z * c + y * stheta,
+      y * x * c + z * stheta, ctheta + y * y * c, y * z * c - x * stheta,
+      z * x * c - y * stheta, z * y * c + x * stheta, ctheta + z * z * c;
+
+  return R;
+};
 
 template <typename Derived>
-DLLEXPORT Eigen::Matrix<typename Derived::Scalar, 3, 1> axis2rpy(const Eigen::MatrixBase<Derived>& a);
+Eigen::Matrix<typename Derived::Scalar, 3, 1> axis2rpy(const Eigen::MatrixBase<Derived>& a) {
+  EIGEN_STATIC_ASSERT_VECTOR_SPECIFIC_SIZE(Eigen::MatrixBase<Derived>, 4);
+  return quat2rpy(axis2quat(a));
+};
 
 /*
  * expmap2x
  */
 template <typename Derived>
-DLLEXPORT GradientVar<typename Derived::Scalar, QUAT_SIZE, 1> expmap2quat(const Eigen::MatrixBase<Derived>& q, const int gradient_ordr);
+GradientVar<typename Derived::Scalar, QUAT_SIZE, 1> expmap2quat(const Eigen::MatrixBase<Derived>& v, const int gradient_order) {
+  EIGEN_STATIC_ASSERT_VECTOR_SPECIFIC_SIZE(Eigen::MatrixBase<Derived>, 3);
+  GradientVar<typename Derived::Scalar, QUAT_SIZE, 1> ret(QUAT_SIZE, 1, EXPMAP_SIZE,gradient_order);
+  auto theta = v.norm();
+  if (theta < pow(std::numeric_limits<typename Derived::Scalar>::epsilon(),0.25)) {
+    ret.value() = expmap2quatDegenerate(v, theta);
+    if(gradient_order>0)
+    {
+      ret.gradient().value() = dexpmap2quatDegenerate(v, theta);
+      if(gradient_order>1)
+      {
+        ret.gradient().gradient().value() = ddexpmap2quatDegenerate(v, theta);
+        if(gradient_order>2)
+        {
+          throw std::runtime_error("expmap2quat does not support gradient order larger than 2");
```

Comment:
Just FYI, I didn't write this code, just moved it from the source file to the header.
I think you're right. However, if we're going to be getting rid of GradientVar and our user gradients completely and relying totally on AutoDiffScalar, this code will be removed in the process.
I can change it for now.